diff --git a/config.def.h b/config.def.h
index 779308a..c9302f8 100644
--- a/config.def.h
+++ b/config.def.h
@@ -130,8 +130,7 @@ static const char *colorname[] = {
   // <<<<<<<<<<<<<<<<<<<< selection-colors
 
   // >>>>>>>>>>>>>>>>>>>> alpha
-  // ==================== alpha
-  // ==================== alpha
+  "black",
   // <<<<<<<<<<<<<<<<<<<< alpha
 };
 
@@ -514,8 +513,8 @@ static char ascii_printable[] =
   "`abcdefghijklmnopqrstuvwxyz{|}~";
 
 // >>>>>>>>>>>>>>>>>>>> alpha
-// ==================== alpha
-// ==================== alpha
+/* bg opacity */
+float alpha = 0.8;
 // <<<<<<<<<<<<<<<<<<<< alpha
 
 // >>>>>>>>>>>>>>>>>>>> focus
diff --git a/config.mk b/config.mk
index 64e00fb..39a72a1 100644
--- a/config.mk
+++ b/config.mk
@@ -14,9 +14,7 @@ PKG_CONFIG = pkg-config
 
 
 # >>>>>>>>>>>>>>>>>>>> alpha
-# ==================== alpha
-ALPHA_OPTS =
-# ==================== alpha
+ALPHA_OPTS = -lXrender
 # <<<<<<<<<<<<<<<<<<<< alpha
 
 
diff --git a/st.1 b/st.1
index 53c4445..8580d00 100644
--- a/st.1
+++ b/st.1
@@ -23,8 +23,8 @@ st \- simple terminal
 .RB [ \-w
 .IR windowid ]
 \" >>>>>>>>>>>>>>>>>>>> alpha
-\" ==================== alpha
-\" ==================== alpha
+.RB [ \-A
+.IR opacity ]
 \" <<<<<<<<<<<<<<<<<<<< alpha
 \" >>>>>>>>>>>>>>>>>>>> working-dir
 \" ==================== working-dir
@@ -53,8 +53,8 @@ st \- simple terminal
 .RB [ \-w
 .IR windowid ]
 \" >>>>>>>>>>>>>>>>>>>> alpha
-\" ==================== alpha
-\" ==================== alpha
+.RB [ \-A
+.IR opacity ]
 \" <<<<<<<<<<<<<<<<<<<< alpha
 \" >>>>>>>>>>>>>>>>>>>> working-dir
 \" ==================== working-dir
@@ -140,8 +140,9 @@ This option is only intended for compatibility,
 and all the remaining arguments are used as a command
 even without it.
 \" >>>>>>>>>>>>>>>>>>>> alpha
-\" ==================== alpha
-\" ==================== alpha
+.B \-A
+allows users to change the opacity of the background
+.TP
 \" <<<<<<<<<<<<<<<<<<<< alpha
 \" >>>>>>>>>>>>>>>>>>>> working-dir
 \" ==================== working-dir
diff --git a/st.h b/st.h
index adbc7af..b8b18a7 100644
--- a/st.h
+++ b/st.h
@@ -137,8 +137,7 @@ extern unsigned int defaultfg;
 extern unsigned int defaultbg;
 
 // >>>>>>>>>>>>>>>>>>>> alpha
-// ==================== alpha
-// ==================== alpha
+extern float alpha;
 // <<<<<<<<<<<<<<<<<<<< alpha
 
 // >>>>>>>>>>>>>>>>>>>> focus
diff --git a/x.c b/x.c
index a9fa781..4209276 100644
--- a/x.c
+++ b/x.c
@@ -125,8 +125,7 @@ typedef struct {
   int l, t;    /* left and top offset */
   int gm;      /* geometry mask */
   // >>>>>>>>>>>>>>>>>>>> alpha
-  // ==================== alpha
-  // ==================== alpha
+  int depth;   /* bit depth */
   // <<<<<<<<<<<<<<<<<<<< alpha
 } XWindow;
 
@@ -269,8 +268,7 @@ static char *opt_line = NULL;
 static char *opt_name = NULL;
 static char *opt_title = NULL;
 // >>>>>>>>>>>>>>>>>>>> alpha
-// ==================== alpha
-// ==================== alpha
+static char *opt_alpha = NULL;
 // <<<<<<<<<<<<<<<<<<<< alpha
 // >>>>>>>>>>>>>>>>>>>> focus
 // ==================== focus
@@ -734,9 +732,7 @@ void xresize(int col, int row) {
       win.w,
       win.h,
       // >>>>>>>>>>>>>>>>>>>> alpha
-      // ==================== alpha
-      DefaultDepth(xw.dpy, xw.scr)
-      // ==================== alpha
+      xw.depth
       // <<<<<<<<<<<<<<<<<<<< alpha
     );
   XftDrawChange(xw.draw, xw.buf);
@@ -805,8 +801,12 @@ void xloadcols(void) {
     }
 
   // >>>>>>>>>>>>>>>>>>>> alpha
-  // ==================== alpha
-  // ==================== alpha
+  /* set alpha value of bg color */
+  if (opt_alpha)
+    alpha = strtof(opt_alpha, NULL);
+  dc.col[defaultbg].color.alpha = (unsigned short)(0xffff * alpha);
+  dc.col[defaultbg].pixel &= 0x00FFFFFF;
+  dc.col[defaultbg].pixel |= (unsigned char)(0xff * alpha) << 24;
   // <<<<<<<<<<<<<<<<<<<< alpha
 
   // >>>>>>>>>>>>>>>>>>>> focus
@@ -1111,8 +1111,8 @@ void xinit(int cols, int rows) {
   // <<<<<<<<<<<<<<<<<<<< swap-mouse
 
   // >>>>>>>>>>>>>>>>>>>> alpha
-  // ==================== alpha
-  // ==================== alpha
+  XWindowAttributes attr;
+  XVisualInfo vis;
   // <<<<<<<<<<<<<<<<<<<< alpha
 
   // >>>>>>>>>>>>>>>>>>>> xresources
@@ -1123,9 +1123,16 @@ void xinit(int cols, int rows) {
   // <<<<<<<<<<<<<<<<<<<< xresources
   xw.scr = XDefaultScreen(xw.dpy);
   // >>>>>>>>>>>>>>>>>>>> alpha
-  // ==================== alpha
-  xw.vis = XDefaultVisual(xw.dpy, xw.scr);
-  // ==================== alpha
+  if (!(opt_embed && (parent = strtol(opt_embed, NULL, 0)))) {
+    parent = XRootWindow(xw.dpy, xw.scr);
+    xw.depth = 32;
+  } else {
+    XGetWindowAttributes(xw.dpy, parent, &attr);
+    xw.depth = attr.depth;
+  }
+
+  XMatchVisualInfo(xw.dpy, xw.scr, xw.depth, TrueColor, &vis);
+  xw.vis = vis.visual;
   // <<<<<<<<<<<<<<<<<<<< alpha
 
   /* font */
@@ -1137,9 +1144,7 @@ void xinit(int cols, int rows) {
 
   /* colors */
   // >>>>>>>>>>>>>>>>>>>> alpha
-  // ==================== alpha
-  xw.cmap = XDefaultColormap(xw.dpy, xw.scr);
-  // ==================== alpha
+  xw.cmap = XCreateColormap(xw.dpy, parent, xw.vis, None);
   // <<<<<<<<<<<<<<<<<<<< alpha
   xloadcols();
 
@@ -1166,10 +1171,6 @@ void xinit(int cols, int rows) {
   xw.attrs.colormap = xw.cmap;
 
   // >>>>>>>>>>>>>>>>>>>> alpha
-  // ==================== alpha
-  if (!(opt_embed && (parent = strtol(opt_embed, NULL, 0))))
-    parent = XRootWindow(xw.dpy, xw.scr);
-  // ==================== alpha
   // <<<<<<<<<<<<<<<<<<<< alpha
   xw.win = XCreateWindow(
       xw.dpy,
@@ -1180,9 +1181,7 @@ void xinit(int cols, int rows) {
       win.h,
       0,
       // >>>>>>>>>>>>>>>>>>>> alpha
-      // ==================== alpha
-      XDefaultDepth(xw.dpy, xw.scr),
-      // ==================== alpha
+      xw.depth,
       // <<<<<<<<<<<<<<<<<<<< alpha
       InputOutput,
       xw.vis,
@@ -1193,11 +1192,8 @@ void xinit(int cols, int rows) {
   memset(&gcvalues, 0, sizeof(gcvalues));
   gcvalues.graphics_exposures = False;
   // >>>>>>>>>>>>>>>>>>>> alpha
-  // ==================== alpha
-  dc.gc = XCreateGC(xw.dpy, parent, GCGraphicsExposures, &gcvalues);
-  xw.buf =
-    XCreatePixmap(xw.dpy, xw.win, win.w, win.h, DefaultDepth(xw.dpy, xw.scr));
-  // ==================== alpha
+  xw.buf = XCreatePixmap(xw.dpy, xw.win, win.w, win.h, xw.depth);
+  dc.gc = XCreateGC(xw.dpy, xw.buf, GCGraphicsExposures, &gcvalues);
   // <<<<<<<<<<<<<<<<<<<< alpha
   XSetForeground(xw.dpy, dc.gc, dc.col[defaultbg].pixel);
   XFillRectangle(xw.dpy, xw.buf, dc.gc, 0, 0, win.w, win.h);
@@ -2227,8 +2223,9 @@ int main(int argc, char *argv[]) {
     die("%s " VERSION "\n", argv0);
     break;
   // >>>>>>>>>>>>>>>>>>>> alpha
-  // ==================== alpha
-  // ==================== alpha
+  case 'A':
+    opt_alpha = EARGF(usage());
+    break;
   // <<<<<<<<<<<<<<<<<<<< alpha
   // >>>>>>>>>>>>>>>>>>>> working-dir
   // ==================== working-dir
