diff --git a/x.c b/x.c
index 93cf5c7..b96e323 100644
--- a/x.c
+++ b/x.c
@@ -1376,11 +1376,6 @@ void xdrawglyphfontspecs(const XftGlyphFontSpec *specs, Glyph base, int len,
   }
 
   // >>>>>>>>>>>>>>>>>>>> bold-is-not-bright
-  // ==================== bold-is-not-bright
-  /* Change basic system colors [0-7] to bright system colors [8-15] */
-  if ((base.mode & ATTR_BOLD_FAINT) == ATTR_BOLD && BETWEEN(base.fg, 0, 7))
-    fg = &dc.col[base.fg + 8];
-  // ==================== bold-is-not-bright
   // <<<<<<<<<<<<<<<<<<<< bold-is-not-bright
 
   if (IS_SET(MODE_REVERSE)) {
