diff --git a/config.def.h b/config.def.h
index dac7b2c..bca4b8c 100644
--- a/config.def.h
+++ b/config.def.h
@@ -299,9 +299,7 @@ static Key key[] = {
  { XK_KP_Delete,     ShiftMask,      "\033[3;2~",    +1,    0},
  { XK_KP_Delete,     XK_ANY_MOD,     "\033[P",       -1,    0},
   // >>>>>>>>>>>>>>>>>>>> delkey
-  // ==================== delkey
-  { XK_KP_Delete,     XK_ANY_MOD,     "\033[3~",      +1,    0},
-  // ==================== delkey
+  { XK_KP_Delete,     XK_ANY_MOD,     "\177",         +1,    0},
   // <<<<<<<<<<<<<<<<<<<< delkey
  { XK_KP_Multiply,   XK_ANY_MOD,     "\033Oj",       +2,    0},
  { XK_KP_Add,        XK_ANY_MOD,     "\033Ok",       +2,    0},
@@ -371,10 +369,7 @@ static Key key[] = {
  { XK_Delete,        ShiftMask,      "\033[3;2~",    +1,    0},
  { XK_Delete,        XK_ANY_MOD,     "\033[P",       -1,    0},
   // >>>>>>>>>>>>>>>>>>>> delkey
-  // ==================== delkey
-  { XK_Delete,        XK_ANY_MOD,     "\033[3~",      +1,    0},
-  { XK_BackSpace,     XK_NO_MOD,      "\177",          0,    0},
-  // ==================== delkey
+  { XK_Delete,        XK_ANY_MOD,     "\177",         +1,    0},
   // <<<<<<<<<<<<<<<<<<<< delkey
  { XK_BackSpace,     Mod1Mask,       "\033\177",      0,    0},
  { XK_Home,          ShiftMask,      "\033[2J",       0,   -1},
