diff --git a/config.def.h b/config.def.h
index 7007edd..5b41392 100644
--- a/config.def.h
+++ b/config.def.h
@@ -104,37 +104,33 @@ unsigned int tabspaces = 8;
 /* Terminal colors (16 first used in escape sequence) */
 static const char *colorname[] = {
   // >>>>>>>>>>>>>>>>>>>> iceberg-theme-dracula-theme-gruvbox-theme-solarized-theme-nord-theme
-  // ==================== iceberg-theme-dracula-theme-gruvbox-theme-solarized-theme-nord-theme
   /* 8 normal colors */
-  "black",
-  "red3",
-  "green3",
-  "yellow3",
-  "blue2",
-  "magenta3",
-  "cyan3",
-  "gray90",
+  "#3b4252", /* black   */
+  "#bf616a", /* red     */
+  "#a3be8c", /* green   */
+  "#ebcb8b", /* yellow  */
+  "#81a1c1", /* blue    */
+  "#b48ead", /* magenta */
+  "#88c0d0", /* cyan    */
+  "#e5e9f0", /* white   */
 
   /* 8 bright colors */
-  "gray50",
-  "red",
-  "green",
-  "yellow",
-  "#5c5cff",
-  "magenta",
-  "cyan",
-  "white",
+  "#4c566a", /* black   */
+  "#bf616a", /* red     */
+  "#a3be8c", /* green   */
+  "#ebcb8b", /* yellow  */
+  "#81a1c1", /* blue    */
+  "#b48ead", /* magenta */
+  "#8fbcbb", /* cyan    */
+  "#eceff4", /* white   */
 
   [255] = 0,
-  // ==================== iceberg-theme-dracula-theme-gruvbox-theme-solarized-theme-nord-theme
   // <<<<<<<<<<<<<<<<<<<< iceberg-theme-dracula-theme-gruvbox-theme-solarized-theme-nord-theme
 
   // >>>>>>>>>>>>>>>>>>>> selection-colors-nord-theme
-  // ==================== selection-colors-nord-theme
   /* more colors can be added after 255 to use with DefaultXX */
-  "#cccccc",
-  "#555555",
-  // ==================== selection-colors-nord-theme
+  "#2e3440", /* background */
+  "#d8dee9", /* foreground */
   // <<<<<<<<<<<<<<<<<<<< selection-colors-nord-theme
 
   // >>>>>>>>>>>>>>>>>>>> alpha
@@ -145,16 +141,14 @@ static const char *colorname[] = {
 
 
 // >>>>>>>>>>>>>>>>>>>> iceberg-theme-dracula-theme-gruvbox-theme-solarized-theme-nord-theme
-// ==================== iceberg-theme-dracula-theme-gruvbox-theme-solarized-theme-nord-theme
 /*
  * Default colors (colorname index)
  * foreground, background, cursor, reverse cursor
  */
-unsigned int defaultfg = 7;
-unsigned int defaultbg = 0;
-static unsigned int defaultcs = 256;
-static unsigned int defaultrcs = 257;
-// ==================== iceberg-theme-dracula-theme-gruvbox-theme-solarized-theme-nord-theme
+unsigned int defaultfg = 257;
+unsigned int defaultbg = 256;
+static unsigned int defaultcs = 257;
+static unsigned int defaultrcs = 256;
 // <<<<<<<<<<<<<<<<<<<< iceberg-theme-dracula-theme-gruvbox-theme-solarized-theme-nord-theme
 
 // >>>>>>>>>>>>>>>>>>>> selection-colors
