diff --git a/config.def.h b/config.def.h
index 2bdfaba..8c40d48 100644
--- a/config.def.h
+++ b/config.def.h
@@ -130,11 +130,9 @@ static const char *colorname[] = {
   // <<<<<<<<<<<<<<<<<<<< iceberg-theme-dracula-theme-gruvbox-theme-solarized-theme-nord-theme
 
   // >>>>>>>>>>>>>>>>>>>> selection-colors-nord-theme
-  // ==================== selection-colors-nord-theme
   /* more colors can be added after 255 to use with DefaultXX */
   "#cccccc",
-  "#555555",
-  // ==================== selection-colors-nord-theme
+  "#2e3440",
   // <<<<<<<<<<<<<<<<<<<< selection-colors-nord-theme
 
   // >>>>>>>>>>>>>>>>>>>> alpha
@@ -158,8 +156,12 @@ static unsigned int defaultrcs = 257;
 // <<<<<<<<<<<<<<<<<<<< iceberg-theme-dracula-theme-gruvbox-theme-solarized-theme-nord-theme
 
 // >>>>>>>>>>>>>>>>>>>> selection-colors
-// ==================== selection-colors
-// ==================== selection-colors
+/* Colors used for selection */
+unsigned int selectionbg = 257;
+unsigned int selectionfg = 7;
+/* If 0 use selectionfg as foreground in order to have a uniform foreground-color */
+/* Else if 1 keep original foreground-color of each cell => more colors :) */
+static int ignoreselfg = 1;
 // <<<<<<<<<<<<<<<<<<<< selection-colors
 
 // >>>>>>>>>>>>>>>>>>>> blinking-cursor
diff --git a/st.h b/st.h
index 07a228d..48add35 100644
--- a/st.h
+++ b/st.h
@@ -46,8 +46,7 @@ enum glyph_attribute {
   // ==================== ligatures
   // <<<<<<<<<<<<<<<<<<<< ligatures
   // >>>>>>>>>>>>>>>>>>>> selection-colors
-  // ==================== selection-colors
-  // ==================== selection-colors
+  ATTR_SELECTED   = 1 << 12,
   // <<<<<<<<<<<<<<<<<<<< selection-colors
   ATTR_BOLD_FAINT = ATTR_BOLD | ATTR_FAINT,
 };
diff --git a/x.c b/x.c
index e0ebaba..ccfce59 100644
--- a/x.c
+++ b/x.c
@@ -1589,8 +1589,11 @@ void xdrawglyphfontspecs(const XftGlyphFontSpec *specs, Glyph base, int len,
   }
 
   // >>>>>>>>>>>>>>>>>>>> selection-colors
-  // ==================== selection-colors
-  // ==================== selection-colors
+  if (base.mode & ATTR_SELECTED) {
+    bg = &dc.col[selectionbg];
+    if (!ignoreselfg)
+      fg = &dc.col[selectionfg];
+  }
   // <<<<<<<<<<<<<<<<<<<< selection-colors
 
   if (base.mode & ATTR_BLINK && win.mode & MODE_BLINK)
@@ -1696,9 +1699,7 @@ void xdrawcursor(
   /* remove the old cursor */
   if (selected(ox, oy))
     // >>>>>>>>>>>>>>>>>>>> selection-colors
-    // ==================== selection-colors
-    og.mode ^= ATTR_REVERSE;
-    // ==================== selection-colors
+    og.mode ^= ATTR_SELECTED;
     // <<<<<<<<<<<<<<<<<<<< selection-colors
 
   // >>>>>>>>>>>>>>>>>>>> ligatures
@@ -1718,33 +1719,17 @@ void xdrawcursor(
   if (IS_SET(MODE_REVERSE)) {
     g.mode |= ATTR_REVERSE;
     // >>>>>>>>>>>>>>>>>>>> selection-colors
-    // ==================== selection-colors
-    // ==================== selection-colors
+    g.fg = defaultcs;
     // <<<<<<<<<<<<<<<<<<<< selection-colors
     g.bg = defaultfg;
     // >>>>>>>>>>>>>>>>>>>> selection-colors
-    // ==================== selection-colors
-    if (selected(cx, cy)) {
-      drawcol = dc.col[defaultcs];
-      g.fg = defaultrcs;
-    } else {
-      drawcol = dc.col[defaultrcs];
-      g.fg = defaultcs;
-    }
-    // ==================== selection-colors
+    drawcol = dc.col[defaultrcs];
     // <<<<<<<<<<<<<<<<<<<< selection-colors
   } else {
     // >>>>>>>>>>>>>>>>>>>> selection-colors
-    // ==================== selection-colors
-    if (selected(cx, cy)) {
-      g.fg = defaultfg;
-      g.bg = defaultrcs;
-    } else {
-      g.fg = defaultbg;
-      g.bg = defaultcs;
-    }
-    drawcol = dc.col[g.bg];
-    // ==================== selection-colors
+    g.fg = defaultbg;
+    g.bg = defaultcs;
+    drawcol = dc.col[defaultcs];
     // <<<<<<<<<<<<<<<<<<<< selection-colors
   }
 
@@ -1894,9 +1879,7 @@ void xdrawline(Line line, int x1, int y1, int x2) {
       continue;
     if (selected(x, y1))
       // >>>>>>>>>>>>>>>>>>>> selection-colors
-      // ==================== selection-colors
-      new.mode ^= ATTR_REVERSE;
-      // ==================== selection-colors
+      new.mode ^= ATTR_SELECTED;
       // <<<<<<<<<<<<<<<<<<<< selection-colors
     if (i > 0 && ATTRCMP(base, new)) {
       xdrawglyphfontspecs(specs, base, i, ox, y1);
